app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -83,28 +98,64 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            Window w = getWindow();
+            w.setFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS, WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        }
+
         if (getSupportFragmentManager() != null && getSupportFragmentManager().getBackStackEntryCount() == 0) {
             initFragments();
         }
 
         setSupportActionBar(findViewById(R.id.toolbar));
-        setupDrawer();
+        try {
+            setupDrawer();
+        } catch (Exception e) {
+            ErrorActivity.reportUiError(this, e);
+        }
     }
 
-    private void setupDrawer() {
+    private void setupDrawer() throws Exception {
         final Toolbar toolbar = findViewById(R.id.toolbar);
         drawer = findViewById(R.id.drawer_layout);
         drawerItems = findViewById(R.id.navigation);
 
-        for(StreamingService s : NewPipe.getServices()) {
-            final String title = s.getServiceInfo().getName() +
-                    (ServiceHelper.isBeta(s) ? " (beta)" : "");
-            final MenuItem item = drawerItems.getMenu()
-                    .add(R.id.menu_services_group, s.getServiceId(), 0, title);
-            item.setIcon(ServiceHelper.getIcon(s.getServiceId()));
+        //Tabs
+        int currentServiceId = ServiceHelper.getSelectedServiceId(this);
+        StreamingService service = NewPipe.getService(currentServiceId);
+
+        int kioskId = 0;
+
+        for (final String ks : service.getKioskList().getAvailableKiosks()) {
+            drawerItems.getMenu()
+                    .add(R.id.menu_tabs_group, kioskId, 0, KioskTranslator.getTranslatedKioskName(ks, this))
+                    .setIcon(KioskTranslator.getKioskIcons(ks, this));
+            kioskId ++;
         }
 
-        drawerItems.getMenu().getItem(ServiceHelper.getSelectedServiceId(this)).setChecked(true);
+        drawerItems.getMenu()
+                .add(R.id.menu_tabs_group, -1, 0, R.string.tab_subscriptions)
```

Comment:
These index numbers should be extracted into static variables, since they are used multiple times.